Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1223 @@
+        //        }
+    }
+
+    function createDefaultBillboard(proxy, sourceUri, uriResolver) {
+        var billboard = new BillboardGraphics();
+        billboard.width = BILLBOARD_SIZE;
+        billboard.height = BILLBOARD_SIZE;
+        billboard.scaleByDistance = new NearFarScalar(2414016, 1.0, 1.6093e+7, 0.1);
+        var DEFAULT_ICON = '../../../Build/Cesium/Assets/Textures/maki/marker.png';
+        billboard.image = resolveHref(DEFAULT_ICON, proxy, sourceUri, uriResolver);
+        return billboard;
+    }
+
+    function createDefaultPolygon() {
+        var polygon = new PolygonGraphics();
+        polygon.outline = true;
+        polygon.outlineColor = Color.WHITE;
+        return polygon;
+    }
+
+    function createDefaultLabel() {
+        var label = new LabelGraphics();
+        label.translucencyByDistance = new NearFarScalar(3000000, 1.0, 5000000, 0.0);
+        label.pixelOffset = new Cartesian2(17, 0);
+        label.horizontalOrigin = HorizontalOrigin.LEFT;
+        label.font = '16px sans-serif';
+        label.style = LabelStyle.FILL_AND_OUTLINE;
+        return label;
+    }
+
+    function createDefaultPolyline() {
+        var polyline = new PolylineGraphics();
+        polyline.width = 4;
+        polyline.material = new PolylineOutlineMaterialProperty();
+        polyline.material.color = Color.RED;
+        polyline.material.outlineWidth = 2;
+        polyline.material.outlineColor = Color.BLACK;
+        return polyline;
+    }
+
+    function createDefaultPoint() {
+        var point = new PointGraphics();
+        point.pixelSize = 15;
+        point.color = Color.YELLOW;
+        return point;
+    }
+
+    // This is a list of the Optional Description Information:
+    //  <name> GPS waypoint name of the waypoint
+    //  <cmt> GPS comment of the waypoint
+    //  <desc> Descriptive description of the waypoint
+    //  <src> Source of the waypoint data
+    //  TODO <link> Link (URI/URL) associated with the waypoint
+    //  <type> Type (category) of waypoint
+    var descriptiveInfoTypes = {
+        position : {
+            text : 'Coordinates'
+        },
+        elevation : {
+            text : 'Elevation in meters',
+            tag : 'ele'
+        },
+        time : {
+            text : 'Time',
+            tag : 'time'
+        },
+        name : {
+            text : 'Name',
+            tag : 'name'
+        },
+        comment : {
+            text : 'Comment',
+            tag : 'cmt'
+        },
+        description : {
+            text : 'Description',
+            tag : 'desc'
+        },
+        source : {
+            text : 'Source',
+            tag : 'src'
+        },
+        number : {
+            text : 'GPS track/route number',
+            tag : 'number'
+        },
+        type : {
+            text : 'Type',
+            tag : 'type'
+        }
+
+    };
+    var scratchDiv = document.createElement('div');
+    function processDescription(node, entity, uriResolver) {
```

Comment:
> Should I still remove the coordinates and name from the description block? At the time I wasn't sure what information was description 'worthy'.

Yes, I would remove them.

> As the processed description becomes HTML is there any easy way to make those test cases you've mentioned without having to parse the HTML?

You should be able to just interrogate the resulting HTML programatically to make sure the write data gets created.  Have a look at the KML tests for ideas on how to compare input and output.  It's not exactly the same, but should help.